src/checkers/inference/model/TernaryVariableSlot.java

```diff
@@ -0,0 +1,90 @@
+package checkers.inference.model;
+
+/**
+ * TernaryVariableSlots represent locations whose values depend on two other VariableSlots.
+ *
+ * TernaryVariableSlots are used as the result slot for constraints such as viewpoint adaptation,
+ * least-upper-bounds, arithmetic operations, and joins between variable and refinement variables.
+ * 
+ * TODO: decouple the multi-use of TernaryVariableSlots into separate, distinctly named slots for
+ * each purpose, ie ViewpointAdaptationResultSlot, LeastUpperBoundSlot, ...
+ */
+public class TernaryVariableSlot extends VariableSlot {
```

Comment:
I don't think using `TernaryVariableSlot` is better than using `TernaryVariableSlot` if we don't clearly separate use cases according to whether it's result of viewpoint adaptation, arithmetic operation or least upper bound. For me, this new `TernaryVariableSlot` seems only renaming `TernaryVariableSlot`.